src/quickgui/qgsquickmapcanvasmap.h

```diff
@@ -0,0 +1,172 @@
+#include "qgsquickmapsettings.h"
+
+class QgsMapRendererParallelJob;
+class QgsMapRendererCache;
+class QgsLabelingResults;
+
+/**
+ * \ingroup quick
+ * This class implements a visual Qt Quick Item that does map rendering
+ * according to the current map settings. Client code is expected to use
+ * MapCanvas item rather than using this class directly.
+ *
+ * QgsQuickMapCanvasMap instance internally creates QgsQuickMapSettings in
+ * constructor. The map settings for other QgsQuick components should
+ * be initialized from  QgsQuickMapCanvasMap's mapSettings
+ *
+ * \note QML Type: MapCanvasMap
+ *
+ * \sa QgsQuickMapCanvas
+ *
+ * \since QGIS 3.2
+ */
+class QUICK_EXPORT QgsQuickMapCanvasMap : public QQuickItem
+{
+    Q_OBJECT
+
+    //! map settings. The map settings should be source of map settings for all other components
+    Q_PROPERTY( QgsQuickMapSettings *mapSettings READ mapSettings )
+    //! do not refresh canvas
+    Q_PROPERTY( bool freeze READ freeze WRITE setFreeze NOTIFY freezeChanged )
+    //! QgsMapRendererParallelJob is running
+    Q_PROPERTY( bool isRendering READ isRendering NOTIFY isRenderingChanged )
+
+    /**
+     * Interval in milliseconds after which the map canvas will be updated while a rendering job is ongoing.
+     * This only has an effect if incrementalRendering is activated.
+     * Default is 250 [ms].
+     */
+    Q_PROPERTY( int mapUpdateInterval READ mapUpdateInterval WRITE setMapUpdateInterval NOTIFY mapUpdateIntervalChanged )
+    //! allow increamental rendering - automatic refresh of map canvas while a rendering job is ongoing
+    Q_PROPERTY( bool incrementalRendering READ incrementalRendering WRITE setIncrementalRendering NOTIFY incrementalRenderingChanged )
+
+  public:
+    //! create map canvas map
+    QgsQuickMapCanvasMap( QQuickItem *parent = nullptr );
+    ~QgsQuickMapCanvasMap() = default;
+
+    //! Return map settings associated
+    QgsQuickMapSettings *mapSettings() const;
+
+    //! Return map canvas for Qt Quick scene graph changes
+    virtual QSGNode *updatePaintNode( QSGNode *oldNode, QQuickItem::UpdatePaintNodeData * );
+
+    //! Return whether canvas refresh is frozen
+    bool freeze() const;
+
+    //! Freeze canvas refresh
+    void setFreeze( bool freeze );
+
+    //! Whether renderer job is running
+    bool isRendering() const;
+
+    //! Return current map update interval for incremental rendering
+    int mapUpdateInterval() const;
+
+    //! Set map update interval for incremental rendering
+    void setMapUpdateInterval( int mapUpdateInterval );
+
+    //! Return whether incremental rendering is on
+    bool incrementalRendering() const;
+    //! Set incremental rendering
+    void setIncrementalRendering( bool incrementalRendering );
+
+  signals:
+    //! rendering starting
+    void renderStarting();
+
+    //! canvas refreshed
+    void mapCanvasRefreshed();
+
+    //! freeze changed
+    void freezeChanged();
+
+    //! is rendering changed
```

Comment:
I am not thrilled by these doxygen comments. I know it's not fun to write those for methods and signals, one didn't write himself.

Note, what I normally do is to really think about the *property* and forget about the setter/getter/signal part. That is I'd copy paste for getter/signal of `isRendering` (and setter for others where applicable) a doxygen string like

```
/**
 * The isRendering property is set to true while a rendering job is pending for this map canvas map.
 * It can be used to show a notification icon about an ongoing rendering job.
 * This is a readonly property.
 */
```

Reply:
I am sorry, but writing documentation is not something I feel comfortable with. I tried to improve the docs based on your suggestion where possible. Copy of documentation was done by [\copydoc](https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdcopydoc) where possible